app/uk/gov/hmrc/apinotificationqueue/controller/QueueController.scala

```diff
@@ -40,6 +40,7 @@ class QueueController @Inject()(queueService: QueueService, fieldsService: ApiSu
 
   private val MISSING_CLIENT_ID_ERROR = s"$CLIENT_ID_HEADER_NAME required."
   private val MISSING_BODY_ERROR = s"Body required."
+  private val UUID_REGEX = "^[0-9a-f]{8}-[0-9a-f]{4}-[1-5][0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}$".r
```

Comment:
In my opinion regular expressions are not needed here.
We are over-complicating the logic.

The method `UUID.fromString(X)` throws already a `java.lang.IllegalArgumentException` if the `X` parameter is not a UUID-type String.

You could simply use the Scala `Try` approach by pattern-matching `Success` and `Failure`.
If it fails you could return 400 with your error message.

Reply:
I think having a Regex here is OK as we are expecting a type 4 UUID (I believe) and there are special rules  for that encapsulated in the Regex. Also the default rules are rather lax. I agree with @gokyo that validation should be pulled up as utility. This will help us in writing isolated unit tests for both the validation *and* the controller code (enabling mocking of validation), and lead to cleaner controller code.